src/test/java/org/apache/maven/buildcache/util/LogFileUtils.java

```diff
@@ -60,4 +61,32 @@ public static String findFirstLineContainingTextsInLogs(final Verifier verifier,
 
         return null;
     }
+
+    /**
+     * Find lines matching all the strings given as parameter in the log file attached to a verifier
+     * @param verifier the maven verifier instance
+     * @param texts all the matching strings to find
+     * @return a list of matching strings
+     * @throws VerificationException
+     */
+    public static List<String> findLinesContainingTextsInLogs(final Verifier verifier, final String... texts)
+            throws VerificationException {
+        List<String> lines = verifier.loadFile(verifier.getBasedir(), verifier.getLogFileName(), false);
+        List<String> result = new ArrayList<>();
+        Iterator it = lines.iterator();
+
+        while (it.hasNext()) {
+            String line = verifier.stripAnsi((String) it.next());
+            boolean matches = true;
+            Iterator<String> toMatchIterator = Arrays.stream(texts).iterator();
```

Comment:
Not a critical piece of code, but it is more elegant and efficient to use [StringUtils#containsAny](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#containsAny-java.lang.CharSequence-java.lang.CharSequence...-) and [Stream#anyMatch](https://docs.oracle.com/en%2Fjava%2Fjavase%2F11%2Fdocs%2Fapi%2F%2F/java.base/java/util/stream/Stream.html#anyMatch(java.util.function.Predicate)):
```java
return Files.lines(filePath) // or lines.stream()
    .anyMatch(line -> StringUtils.containsAny(line, texts));
    .
```

Reply:
Done. Slightly changed you proposal since the goal is to test the presence of all strings.